src/data/complex/basic.lean

```diff
@@ -470,11 +470,23 @@ by simp [abs]
 @[simp] lemma abs_mul (z w : ℂ) : abs (z * w) = abs z * abs w :=
 by rw [abs, norm_sq_mul, real.sqrt_mul (norm_sq_nonneg _)]; refl
 
+/-- `complex.abs` as a `monoid_with_zero_hom`. -/
+@[simps] noncomputable def abs_hom : ℂ →*₀ ℝ :=
+{ to_fun := abs,
+  map_zero' := abs_zero,
+  map_one' := abs_one,
+  map_mul' := abs_mul }
+
+-- todo: `map_prod` `fun-like`.
```

Comment:
You mean #12354, which is merged?
```suggestion
```

Reply:
ooh, amazing! I wasn't aware of this, it was just not working well in the Wedderburn PR, but I hadn't merged master for a bit.